File: public/geojson/data_prep.py
import json
import pandas as pd

def flatten_json(json):
    data = map(lambda x: x['properties'], json['features'])
    return pd.DataFrame(data)

def main(filePath):
    with open(filePath, 'r') as f:
        data = json.loads(f.read())
    df = flatten_json(data)
    df.to_csv('countries-prop.csv')



if __name__ == "__main__":

    main('countries.geojson')

File: public/geojson/dataPrep.js
const fs = require('fs');
const csv = require('csv-parse/lib/sync');

const records = csv(fs.readFileSync('mappings.csv'), {
    columns: true,
    skip_empty_lines: true
  })


mapJSON('countries.geojson', records)


function cleanJSON(filePath, oldName, newName) {
    const data = fs.readFileSync(filePath)
    const json = JSON.parse(data)

    json.features = json.features.map(x => {
        x.properties[newName] = x.properties[oldName]
        delete x.properties[oldName]
        return x
    })
    fs.writeFileSync(filePath, JSON.stringify(json, null, 2))
}

function mapJSON(filePath, records) {
    const data = fs.readFileSync(filePath)
    const json = JSON.parse(data)

    json.features = json.features.map(x => {
        const match = records.filter(record => record['From'] === x.properties.name);
        if (match && match.length) {
            x.properties.name = match[0]['To']
        }   
        return x
    });
    fs.writeFileSync(filePath, JSON.stringify(json, null, 2))
}

